sdk/src/subnet_id.rs

```diff
@@ -1,311 +1,310 @@
+use fil_actors_runtime::cbor;
+use fnv::FnvHasher;
 use fvm_shared::address::Address;
 use lazy_static::lazy_static;
 use serde_tuple::{Deserialize_tuple, Serialize_tuple};
 use std::fmt;
-use std::path::{Path, PathBuf};
+use std::hash::{Hash, Hasher};
 use std::str::FromStr;
 
 use crate::error::Error;
-use crate::set_network_from_env;
 
+const SUBNET_ERR_TAG: &str = "subnetID";
+/// MaxChainID is the maximum chain ID value
+/// possible.
+const MAX_CHAIN_ID: u64 = 0xffffffffffffffff;
+
+/// SubnetID is a unique identifier for a subnet.
+/// It is composed of the chainID of the root network, and the address of
+/// all the subnet actors from the root to the corresponding level in the
+/// hierarchy where the subnet is spawned.
 #[derive(PartialEq, Eq, Hash, Clone, Debug, Serialize_tuple, Deserialize_tuple)]
 pub struct SubnetID {
-    parent: String,
-    actor: Address,
+    root: u64,
```

Comment:
Unfortunately, `ChainID` is not cbor-serializable, so to avoid having to do a lot of translations or having to implement my own serialization out-of-band I decided to stick to `u64`. But I guess I can change it if it is annoying for you.

Reply:
Yeah, I agree on that part, it's an annoying data type to use.